libraries/CurieSoftwareSerial/SoftwareSerial.cpp

```diff
@@ -94,14 +94,21 @@ bool SoftwareSerial::listen()
     bufferOverflow = false;
     _receive_buffer_head = _receive_buffer_tail = 0;
     active_object = this;
-    rxPin = _receivePin;
+    _rxPin = _receivePin;
+    rxIntraBitDelay = _rx_delay_intrabit;
+    rxCenteringDelay = _rx_delay_centering;
```

Comment:
One other way, equally not pretty, to allow static ISR routine to access variable of an object is,
1.  Define the ISR as private in a Class.
2.  Instantiate a static object of the Class.
3.  Define a static function, ISR_wrapper, that the sole purpose is to call the ISR of the Class via the static object.
The only advantage here is that the ISR is now free to access all variables in the Class without making them global, static, etc.